packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -456,6 +456,29 @@ export class DataStores implements IDisposable {
         return builder.getGCData();
     }
 
+    /**
+     * After GC has run, called to notify this Container's data stores of routes that are used in it.
+     * @param usedRoutes - The routes that are used in all data stores in this Container.
+     */
+    public updateUsedRoutes(usedRoutes: string[]) {
+        // Build a map of data store ids to routes used in it.
+        const usedRoutesMap: Map<string, string[]> = new Map();
+        for (const route of usedRoutes) {
+            const dataStoreId = route.split("/")[1];
+            const dataStoreRoute = route.slice(dataStoreId.length + 1);
+            if (usedRoutesMap.has(dataStoreId)) {
+                usedRoutesMap.get(dataStoreId)?.push(dataStoreRoute);
```

Comment:
nit: Can avoid the slight weirdness of `?.` by doing `const routes = usedRoutesMap.get(dataStoreId);` first, then checking if `routes` is undefined instead of using `.has()`.

Reply:
Yeah, I went back and forth on this. I wish the compiler would realize that I already checked that the map has this data before getting it :-).